src/qililab/qprogram/quantum_machines_compiler.py

```diff
@@ -259,6 +259,11 @@ def _handle_for_loop(self, element: ForLoop):
         if element.variable.domain is Domain.Time:
             start = max(start, self.MINIMUM_TIME)
 
+        if element.variable.domain is Domain.Voltage or (
+            element.variable.domain is Domain.Scalar and isinstance(element.variable, FloatVariable)
+        ):
+            stop += step / 2
```

Comment:
Do you want to check if the variable is of `float` type? Then you can do it directly with either `isinstance(element.variable, FloatVariable)` or `isinstance(element.variable, float)`. No need to check the domains.

Reply:
True I made everything more complicated, I started with Voltage and realized that float scalar should be included. Fixed it!

Reply:
@fedonman As you can see in the tests, the Frequencies also are included in the condition FloatVariable and so the stop is also modified even though we don't want to include it inside the condition.

This test error only appears because frequency is considered a Float:
![image](https://github.com/user-attachments/assets/3e71dd64-3a30-4f7c-9c8b-b3329d4de22f)